### ISSUE TITLE
Cannot use Stack and libsodium

### ISSUE THREAD

**Issue by matthew-piziak** [NixOS/nixpkgs#59374]
## Issue description

On Ubuntu, installing `libsodium-dev` and building a Stack project that depends on it works as expected. On NixOS it fails with `* Missing (or bad) C library: sodium` even if the libs and includes are specified with `--extra-lib-dirs=/nix/store/n6cnnyj7fq8rislwq1557f00ylpvbk9a-libsodium-1.0.16-dev/lib --extra-include-dirs=/nix/store/n6cnnyj7fq8rislwq1557f00ylpvbk9a-libsodium-1.0.16-dev/include`

### Steps to reproduce

Create a Stack project with `sodium` as a dependency. Install `unstable.libsodium.dev` in a `nix-shell`'s `buildInputs`. Run `stack install` with `--extra-lib-dirs=/nix/store/n6cnnyj7fq8rislwq1557f00ylpvbk9a-libsodium-1.0.16-dev/lib --extra-include-dirs=/nix/store/n6cnnyj7fq8rislwq1557f00ylpvbk9a-libsodium-1.0.16-dev/include`. Note the error:
```
Cabal-simple_mPHDZzAJ_2.2.0.1_ghc-8.4.4: Missing dependency on a foreign
    library:
    * Missing (or bad) C library: sodium
    This problem can usually be solved by installing the system package that
    provides this library (you may need the "-dev" version). If the library is
    already installed but in a non-standard location then you can use the flags
    --extra-include-dirs= and --extra-lib-dirs= to specify where it is.If the
    library file does exist, it may contain errors that are caught by the C
    compiler at the preprocessing stage. In this case you can re-run configure
    with the verbosity flag -v3 to see the error messages.
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.32, NixOS, 18.09.2505.46d3867a08a (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixpkgs-18.09.2505.46d3867a08a"`
 - channels(matt): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by matthewbauer** [comment]
The lib-dir value looks wrong

**Comment by matthew-piziak** [comment]
@matthewbauer It does? What looks wrong about it? I've checked and the path does exist on my file system...

**Comment by matthew-piziak** [comment]
@matthewbauer Why does the `lib-dirs` value look wrong?

**Comment by matthewbauer** [comment]
Sorry- the lib-dirs directory doesn’t have the right libs. If you look in the directory there id no .a or .so files. You need libsodium.out for this case.

**Comment by matthew-piziak** [comment]
You're right...I used `--extra-lib-dirs=/nix/store/kgq1fxb7c84r212kmq4p8zcxyi7fv51r-libsodium-1.0.16/lib --extra-include-dirs=/nix/store/n6cnnyj7fq8rislwq1557f00ylpvbk9a-libsodium-1.0.16-dev/include` and that worked for me. I had to install both `out` and `dev` like this: 
```
buildInputs = [ pkgs.libsodium.dev pkgs.libsodium.out ];
```

Thank you for the assistance! That said...does this make sense to you? It seems pretty confusing to me that I'd have to install two packages.
